File: 0496-next-greater-element-i/0496-next-greater-element-i.cpp
class Solution {
public:
    vector<int> nextGreaterElement(vector<int>& nums1, vector<int>& nums2) {
        vector<int> res(nums1.size(), -1);

        for (int i = 0; i < nums1.size(); i++) {
            bool found = false;
            for (int j = 0; j < nums2.size(); j++) {
               
                if (nums2[j] == nums1[i]) {
                    for (int k = j + 1; k < nums2.size(); k++) {
                        if (nums2[k] > nums1[i]) {
                            res[i] = nums2[k];
                            found = true;
                            break;
                        }
                    }
                    break;
                }
            }
        }
        return res;
    }
};

File: 0496-next-greater-element-i/0496-next-greater-element-i.py
class Solution:
    def nextGreaterElement(self, nums1: List[int], nums2: List[int]) -> List[int]:
        stack=[]
        freq={}
        for i in range(len(nums2)-1,-1,-1):
            while stack and stack[-1]<=nums2[i]:
                stack.pop()
            if not stack:
                freq[nums2[i]]=-1
            else:
                freq[nums2[i]]=stack[-1]
            stack.append(nums2[i])
        return [freq[num] for num in nums1]
            
        
        